### ISSUE TITLE
Validator rules not converting fields to underscore

### ISSUE THREAD

**Issue by half2me** [cloudcreativity/laravel-json-api#471]
In my validator I have the following rules:
```php
return [
	'something' => 'unique:mytable', # works
	'something-else' => 'unique:mytable', # doesn't work
];
```

I get an sql error about the column: `something-else` not existing. I can fix it by manually editing the rule to: `'something-else' => 'unique:mytable:something_else'` but I have to do this for every field that is composed of multiple words. Is this the way its supposed to work, or can we automatically run `Str::underscore()` on these?

**Comment by half2me** [comment]
Also the documentation conveniently leaves out field names which are composed of multiple words, and this makes it very difficult to know what to do. I find most of my issues with jsonAPI stem from converting between camelCase, snake-case and under_score. Could we have a few examples in the docs covering these, as longer field names are going to show up in every project eventually.

**Comment by lindyhopchris** [comment]
There's no processing of the rules - they're passed directly into the Laravel validator. So yes, you'd need to do `unique:mytable:something_else` as you say to get it working. It makes sense to do this because `something-else` refers to the input from the client, whereas the `unique` takes settings on how to query the database.

I take your point about the docs for converting keys... feel free to submit any PRs adding examples of key conversion; otherwise I'll add in when I next review the docs as it's a good suggestion!

**Comment by half2me** [comment]
Okay, makes sense. Yeah, I can submit a PR for the docs.

**Comment by lindyhopchris** [comment]
Closed via #472
